alertmanager/mask.go

```diff
@@ -0,0 +1,72 @@
+// promconfig
+// Copyright 2020 Percona LLC
+//
+// Based on Prometheus systems and service monitoring server.
+// Copyright 2015 The Prometheus Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alertmanager
+
+import (
+	"reflect"
+	"strconv"
+)
+
+const maskedValue = "xxxxxxxx"
+
+// MaskSensitiveData loops over Config struct and masks sensitive data.
+func MaskSensitiveData(c interface{}) {
```

Comment:
Too bad that this function modifies original object. Maybe we can create method for `Config`, something like:
```
func (c *Config) Mask *Config {
    nc := c.Copy()
    MaskSensitiveData(nc)
    return nc
}
```

In the end we don't need generic solution. What do you think?

Reply:
Yep. I had the same concern about it. Moving the code to the root directory and implementing wrappers for `alertmanager.Config` and `promconfig.Config` would be best.